src/components/Search/SearchComponent.tsx

```diff
@@ -83,7 +83,7 @@ function SearchComponent({
     },
   });
 
-  const [formIsDisplayed, setFormIsDisplayed] = useState(!hasNonEditableState);
+  const [formIsDisplayed, setFormIsDisplayed] = useState(!hasEditableState);
```

Comment:
if you invert the meaning, shouldn't you also change how you use it? In this case removing the `!`?